src/components/parcels/ParcelForm.svelte

```diff
@@ -0,0 +1,313 @@
+    didParameterDictionariesChange(selectedParmeterDictionaries);
+
+  $: {
+    hasPermission =
+      mode === 'edit'
+        ? featurePermissions.parcels.canUpdate(user, { owner: parcelOwner })
+        : featurePermissions.parcels.canCreate(user);
+    permissionError = `You do not have permission to ${mode === 'edit' ? 'edit this' : 'create a'} parcel.`;
+    pageTitle = mode === 'edit' ? 'Parcel' : 'New Parcel';
+    pageSubtitle = mode === 'edit' ? savedParcelName : '';
+    saveButtonText = mode === 'edit' && !parcelModified ? 'Saved' : 'Save';
+  }
+
+  /**
+   * selectedParameterDictionaries keeps track of false values for the table while the saved list doesn't so
+   * we need to do a custom comparison.
+   */
+  function didParameterDictionariesChange(parameterDictionaryMap: Record<number, boolean>): boolean {
+    for (const parameterDictionaryIdString of Object.keys(parameterDictionaryMap)) {
+      const parameterDictionaryId = parseInt(parameterDictionaryIdString);
+
+      if (
+        (selectedParmeterDictionaries[parameterDictionaryId] &&
+          !(parameterDictionaryId in savedParameterDictionaryIds)) ||
+        (!selectedParmeterDictionaries[parameterDictionaryId] && savedParameterDictionaryIds[parameterDictionaryId])
+      ) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  function onToggleChannelDictionary(event: CustomEvent) {
+    parcelChannelDictionaryId = event.detail.id;
+  }
+
+  function onToggleCommandDictionary(event: CustomEvent) {
+    parcelCommandDictionaryId = event.detail.id;
+  }
+
+  function onToggleParameterDictionary(event: CustomEvent) {
+    selectedParmeterDictionaries = event.detail.ids;
+  }
+
+  function onToggleSequenceAdaptation(event: CustomEvent) {
+    parcelSequenceAdaptationId = event.detail.id;
+  }
+
+  async function saveParcelToParameterDictionaries(): Promise<void> {
+    const parcelToParameterDictionariesToAdd: Omit<ParcelToParameterDictionary, 'id'>[] = [];
+    const parcelToParameterDictionaryIdsToDelete: number[] = [];
+
+    Object.keys(selectedParmeterDictionaries).forEach(parameterDictionaryIdString => {
+      const parameterDictionaryId = parseInt(parameterDictionaryIdString);
+      const isSelected = selectedParmeterDictionaries[parameterDictionaryId];
+
+      if (!isSelected && savedParameterDictionaryIds[parameterDictionaryId]) {
+        // Parameter dictionary was removed from the parcel.
+        parcelToParameterDictionaryIdsToDelete.push(parameterDictionaryId);
+      } else if (isSelected && !savedParameterDictionaryIds[parameterDictionaryId] && parcelId) {
+        // Parameter dictionary was freshly added to the parcel and hasn't been saved yet.
+        parcelToParameterDictionariesToAdd.push({
+          parameter_dictionary_id: parameterDictionaryId,
+          parcel_id: parcelId,
+        });
+      }
+    });
+
+    if (parcelToParameterDictionariesToAdd.length > 0) {
```

Comment:
BONUS: You can do this now or defer to the next sprint, but this query can potentially be merged into one. Refer to `UPDATE_CONSTRAINT_MODEL_SPECIFICATIONS` for an example.

Reply:
I'll defer for now, trying to push this through.